src/daemon/engine/CMakeLists.txt

```diff
@@ -58,10 +59,17 @@ add_library(ocvsmd_engine
 )
 target_link_libraries(ocvsmd_engine
         PUBLIC udpard
-        PUBLIC canard
         PUBLIC ${engine_transpiled}
         PUBLIC ocvsmd_common
 )
+if (${PLATFORM_OS_TYPE} STREQUAL "linux")
+    target_sources(ocvsmd_engine
+            PRIVATE platform/can/socketcan.c
+    )
+    target_link_libraries(ocvsmd_engine
+            PUBLIC canard
+    )
```

Comment:
I suggest moving this block outside of the conditional to avoid the false implication that CAN requires Linux